### DISCUSSION TITLE
How do I save residual data?

### DISCUSSION THREAD

**Discussion by HeBe-lab** [phoebe-project/phoebe2#534] [Questions]
Hello, I have got the "Phases Residual" diagram now, how can I save the residual data from this diagram?
<img width="573" alt="截屏2021-09-26 上午10 54 45" src="https://user-images.githubusercontent.com/86087826/134791698-08ff1858-e10a-4eaa-be89-7716e38b93dc.png">

**Comment by kecnry** [comment]
You can access the array by calling [b.calcuate_residuals()](http://phoebe-project.org/docs/2.3/api/phoebe.parameters.ParameterSet.calculate_residuals) (which will be in time-order and the same length as the dataset observation times).  If you also need the array of phases, you can pass your dataset times to [b.to_phase](http://phoebe-project.org/docs/2.3/api/phoebe.frontend.bundle.Bundle.to_phase).

-Kyle

**Reply by HeBe-lab** [reply]
Thank you for your reply.
